### PR TITLE
[Orders][API]  OrderItem removal validation

### PR DESCRIPTION
**Pull request by Rafikooo**
| Q               | A                                                            |
|-----------------|--------------------------------------------------------------|
| Branch?         | 1.12                   |
| Bug fix?        | yes                                                       |
| New feature?    | no                                                      |
| BC breaks?      | no                                                      |
| Deprecations?   | no |
| License         | MIT                                                          |

Fixes:
(in prod env this information was hidden which could confuse the client)
image
After:
image

### CODE REVIEW

File: src/Sylius/Bundle/ApiBundle/Controller/DeleteOrderItemAction.php [hunk 1]
```diff
@@ -14,25 +14,34 @@
...
-        );
+        $orderItemId = $request->attributes->get('itemId');
+        $tokenValue = $request->attributes->get('tokenValue');
 
-        $this->commandBus->dispatch($command);
+        $orderItem = $this->orderItemRepository->findOneByIdAndCartTokenValue($orderItemId, $tokenValue);
+
+        if ($orderItem === null) {
+            throw new OrderItemNotFoundException();
+        }
```
Comment (Rafikooo) [inline src/Sylius/Bundle/ApiBundle/Controller/DeleteOrderItemAction.php:42]: I tried to put this logic inside the command handler, but the exception thrown there is caught by the middleware, which throws another exception that is not mapped to status 422 and request ends with 500. Throwing it here skips the middleware

**Review by NoResponseMate — APPROVED** [review]

**Comment by NoResponseMate** [comment]
Thank you, @Rafikooo!

File: tests/Api/Shop/OrdersTest.php [hunk 2]
```diff
@@ -371,4 +371,24 @@ public function it_removes_item_from_the_cart(): void
...
+        $commandBus = self::getContainer()->get('sylius.command_bus');
+
+        $pickupCartCommand = new PickupCart($tokenValue);
+        $pickupCartCommand->setChannelCode('WEB');
+        $commandBus->dispatch($pickupCartCommand);
+        $nonExistingOrderItemId = 123;
+
+        $this->client->request('DELETE', sprintf('/api/v2/shop/orders/%s/items/%s', $tokenValue, $nonExistingOrderItemId));
+
+        $this->assertResponseCode($this->client->getResponse(), Response::HTTP_UNPROCESSABLE_ENTITY);
```
Comment (coldic3) [inline tests/Api/Shop/OrdersTest.php:392]: This is a case that the order item resource cannot be found. Shouldn't it be 404 instead?

Comment (Rafikooo) [reply tests/Api/Shop/OrdersTest.php:392]: You may be right. I might have overthink it too much, such as the order resource part is correct and its sub-resource is incorrect, hence the status code 422.

Alternatively:
https://www.rfc-editor.org/rfc/rfc4918#section-11.2
"but was unable to process the contained instructions"
Removing something that doesn't exist is unprocessable 💃 

I am not able to clearly state which one is correct 🤔

**Pull request merged**